scene/resources/3d/navigation_mesh_source_geometry_data_3d.cpp

```diff
@@ -30,6 +30,17 @@
 
 #include "navigation_mesh_source_geometry_data_3d.h"
 
+#ifdef DEBUG_ENABLED
+#define NAVMESH_SOURCE_GEOMETRY_PARSE_RENDERING_MESHES_WARN_MSG()                                                                                    \
+	if (!Engine::get_singleton()->is_editor_hint()) {                                                                                                \
+		WARN_PRINT_ONCE("Source geometry parsing for navigation mesh baking had to parse RenderingServer meshes at runtime.\n\
+		This poses a significant performance issues as visual meshes store geometry data on the GPU and transferring this data back to the CPU blocks the rendering.\n\
+		For runtime (re)baking navigation meshes use and parse collision shapes as source geometry or create geometry data procedurally in scripts."); \
+	};
+#else
+#define NAVMESH_SOURCE_GEOMETRY_PARSE_RENDERING_MESHES_WARN_MSG()
+#endif
```

Comment:
Seems to have some trailing `\` alignment issues (with the tab size clang-format uses):
https://github.com/godotengine/godot/actions/runs/8766104374/job/24057840243?pr=90921

It's only used in one place though, so does this needs to be a macro? It could just be inline enclosed in `DEBUG_ENABLED`.

Reply:
True, originally I had multiple places in mind where this needs to be added but then I realized late that all those other functions use the add_mesh() function internally so I switch it over here without replacing the macro.